### PR TITLE
Include optional fields into BayMeds

### PR DESCRIPTION
**Pull request by ChongWeiJie29**
In v1.2, all fields were compulsory. This commit will convert
some of the intended fields to be optional

### CODE REVIEW

**Review by jingyu987 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/prescription/Prescription.java [hunk 1]
```diff
@@ -20,15 +21,15 @@ public class Prescription {
...
-    private final Frequency frequency;
+    private final Optional<Dosage> dosage;
+    private final Optional<Frequency> frequency;
     private final Date startDate;
-    private final Date endDate;
-    private final Date expiryDate;
-    private final Stock totalStock;
+    private final Optional<Date> endDate;
+    private final Optional<Date> expiryDate;
+    private final Optional<Stock> totalStock;
```
Comment (shyanyong) [inline src/main/java/seedu/address/model/prescription/Prescription.java:29]: should total stock be optional?

File: src/main/java/seedu/address/model/prescription/Prescription.java [hunk 2]
```diff
@@ -20,15 +21,15 @@ public class Prescription {
...
-    private final Dosage dosage;
-    private final Frequency frequency;
+    private final Optional<Dosage> dosage;
+    private final Optional<Frequency> frequency;
     private final Date startDate;
-    private final Date endDate;
-    private final Date expiryDate;
-    private final Stock totalStock;
+    private final Optional<Date> endDate;
+    private final Optional<Date> expiryDate;
```
Comment (shyanyong) [inline src/main/java/seedu/address/model/prescription/Prescription.java:28]: Should expiry date also be optional?

File: src/main/java/seedu/address/model/prescription/Prescription.java (hunk 1 above)
Comment (ChongWeiJie29) [reply src/main/java/seedu/address/model/prescription/Prescription.java:29]: I suggest stock and all dates to be optional cuz sometimes users wont know how many they have at 1 go/laze to count and input. Similar for dates, they may not know when the meds will expire/dw to include at first. So they dont need to include these fields when they r first adding the prescription. Then they can use the edit feature whenever they want to add/change these fields

Comment (ChongWeiJie29) [reply src/main/java/seedu/address/model/prescription/Prescription.java:29]: The only mandatory field is name, then the add command will automatically put in a default start date, default consumption count and default is completed field. The medication name, and start date, will be what we use to identify each prescription.

**Review by shyanyong — APPROVED** [review]
LGTM

**Pull request merged**